week-3/todo-list/script.js

```diff
@@ -1,25 +1,68 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
-
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
 
-populateTodoList(todos);
-
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
-  event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+
+  let list = document.getElementById("todo-list");
+  let listValue = list.value;
+ 
+  
+  let li = document.createElement("li")
+  li.classList.add("style-list");
+    list.appendChild(li);
+
+    
+  let container = document.createElement("span")
+    container.classList.add("layout");
+      list.appendChild(container);
+
+      
+    let checkIcon = document.createElement("i");
+    checkIcon.className = 'fa fa-check';
+    checkIcon.setAttribute("aria-hidden", "true");
+    checkIcon.onclick = function() {
+      li.classList.toggle("completed");
+    }
+     container.appendChild(checkIcon);
+   
+
+    let trashIcon = document.createElement("i");
+    trashIcon.className = 'fa fa-trash';
+    trashIcon.setAttribute("aria-hidden", "true");
+    trashIcon.onclick = function(){
+      li.remove();
+   };
+        container.appendChild(trashIcon);
+
+  let textNode = document.createTextNode(listValue);
+  li.appendChild(container);
+  li.appendChild(textNode);
+       
+  const ul = document.getElementById("task-list");
+  ul.appendChild(li);
+
+  list.value = "";
+   event.preventDefault();
 }
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
-  // Write your code here...
+
+function deleteAllCompletedTodos(event) {
+  let todoList = document.getElementById("task-list");
+  let itemsTocompleted = document.querySelectorAll(".completed");
+  itemsTocompleted.forEach(function (item) {
+    todoList.removeChild(item);
+  });
 }
```

Comment:
If you look at your code, you had created the `deleteAllCompletedTodos` function but didn't use it **anywhere**. You need to call it when the corresponding button-`Remove all completed` is clicked.
Also, if you look at your preview when you click the "Remove all completed" button, it deletes everything, even uncompleted tasks.
 But why? 
This happens because the button has a type of "submit", which tries to submit the form and refresh the page. 
To prevent this behaviour, you can add an event listener to the button and call the `deleteAllCompletedTodos` function from there!!!
example:
```
const removeButton = document.getElementById("remove-button");//Assuming the given Id exists.
removeButton.addEventListener("click", deleteAllCompletedTodos);
```